src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1344,9 +1354,48 @@ private static bool IsLessDerivedThanAny<TMember>(int index, TypeSymbol type, Ar
                 {
                     return true;
                 }
-            }
+                else if (currentType.IsExtension)
+                {
+                    if (currentType.ExtendedTypeNoUseSiteDiagnostics is { } extendedType)
+                    {
+                        if (extendedType.Equals(type, TypeCompareKind.ConsiderEverything))
+                        {
+                            return true;
+                        }
 
-            return false;
+                        if (extendedType is TypeParameterSymbol currentTypeParameter)
+                        {
+                            var baseClass = currentTypeParameter.EffectiveBaseClass(ref useSiteInfo);
+                            if (type.Equals(baseClass, TypeCompareKind.ConsiderEverything)
+                                || isLessDerived(type, currentType: baseClass, ref useSiteInfo))
+                            {
+                                return true;
+                            }
+
+                            foreach (var i in currentTypeParameter.AllEffectiveInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteInfo))
+                            {
+                                if (type.Equals(i, TypeCompareKind.ConsiderEverything))
+                                {
+                                    return true;
+                                }
+                            }
+                        }
+                        else if (extendedType.IsInterfaceType())
+                        {
+                            if (extendedType.AllInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteInfo).Contains((NamedTypeSymbol)type))
```

Comment:
If I understand correctly, this will use `TypeCompareKind.AllNullableIgnoreOptions` under the hood but above we use `TypeCompareKind.ConsiderEverything` - is that difference intentional?

Reply:
> If I understand correctly, this will use `TypeCompareKind.AllNullableIgnoreOptions` under the hood but above we use `TypeCompareKind.ConsiderEverything` - is that difference intentional?

FWIW, this is consistent with what we do for interfaces above.

Reply:
Yes, I stuck with existing comparison conventions. FWIW, I couldn't find a test affected by the `ConsiderEverything` on base class (changing it to `AllIgnoreOptions`) and I didn't manage to cook one up either.